### ISSUE TITLE
Incorrect chart: .rules[2].name: field not declared in schema

### ISSUE THREAD

**Issue by batazor** [akuity/kargo#1532]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

# Screenshots

<img width="1354" alt="Screenshot 2024-02-24 at 23 55 49" src="https://github.com/akuity/kargo/assets/1643797/1f8ab489-ebeb-4913-b741-3089152ee9f0">

# Steps to Reproduce

Just install helm chart v0.4.2

# Version

```
v0.4.2
```

# Logs

```

one or more objects failed to apply, reason: failed to create typed patch object (/kargo-management-controller; rbac.authorization.k8s.io/v1, Kind=ClusterRole): 

.rules[2].name: field not declared in schema,failed to create typed patch object (/kargo-webhooks-server; 
rbac.authorization.k8s.io/v1, Kind=ClusterRole): 
.rules[2].name: field not declared in schema (retried 2 times).
```

**Comment by krancour** [comment]
We've seen this before, but it always presented as a warning and didn't block the installation from succeeding.

This is fixed in `main` already.

Is your experience different and it is actually blocking the installation from succeeding?

**Comment by batazor** [comment]
Great. The services don't seem to be affected, although it is alarming

**Comment by krancour** [comment]
> although it is alarming

#1518 has the details.

**Comment by sennerholm** [comment]
It's blocking my argo upgrade with:
```
rbac.authorization.k8s.io/v1/ClusterRole
kargo
kargo-management-controller
 SyncFailed
failed to create typed patch object (/kargo-management-controller; rbac.authorization.k8s.io/v1, Kind=ClusterRole): .rules[2].name: field not declared in schema
```

It would be super if it could get backported to 0.4 chart, to have it failing or warning doesn't give a good first impression :-). I will run with a patched version myself of the chart with the "dex" fix (#1460) in, so we have a workaround for us...
